boilerplate/openshift/golang-codecov/standard.mk

```diff
@@ -0,0 +1,24 @@
+GOOS?=$(shell go env GOOS)
+GOARCH?=$(shell go env GOARCH)
+
+unexport GOFLAGS
+GOFLAGS_MOD ?=
+
+GOENV=GOOS=${GOOS} GOARCH=${GOARCH} CGO_ENABLED=0 GOFLAGS=${GOFLAGS_MOD}
+
+CODECOV_CONVENTION_DIR := boilerplate/openshift/golang-codecov
+
+ifeq ($(origin TESTTARGETS), undefined)
+TESTTARGETS := $(shell ${GOENV} go list -e ./... | egrep -v "/(vendor)/")
+endif
+# ex, -v
+TESTOPTS :=
+
+# coverage: Code coverage analysis and reporting.
+.PHONY: coverage
+coverage:
+	${CODECOV_CONVENTION_DIR}/codecov.sh
+
+.PHONY: go-test
+go-test:
+	${GOENV} go test $(TESTOPTS) $(TESTTARGETS)
```

Comment:
We ought to think ahead to when this convention will be one of many imported by an operator, and `go-test` may be defined in more than one of them. Today in golang-osd-operator it's [aliased](https://github.com/openshift/boilerplate/blob/7d81a9d4ba6ed1b17ed0f0ceee85cb9def9884d4/boilerplate/openshift/golang-osd-operator/standard.mk#L242-L244) to `test`, which is used for CI, and it's one of the targets run by [default](https://github.com/openshift/boilerplate/blob/7d81a9d4ba6ed1b17ed0f0ceee85cb9def9884d4/boilerplate/openshift/golang-osd-operator/standard.mk#L105) (argless `make`). Perhaps we should rename this one `codecov-go-test` to avoid conflicts.

For that matter, we need to think about the source of truth for shared variables like `GOENV`, `TESTOPTS`, and `TESTTARGETS`.

As currently implemented, if you consume multiple conventions, rules/vars will take [precedence](https://github.com/openshift/boilerplate/blob/7d81a9d4ba6ed1b17ed0f0ceee85cb9def9884d4/boilerplate/update#L201) in the reverse of the [order](https://github.com/openshift/boilerplate/blob/7d81a9d4ba6ed1b17ed0f0ceee85cb9def9884d4/boilerplate/update#L173) you list them in your [config](https://github.com/openshift/boilerplate/blob/7d81a9d4ba6ed1b17ed0f0ceee85cb9def9884d4/boilerplate/update#L34). So unless we do something else/different, it's on the consumer to list conventions in ascending order of importance.

Reply:
Yes, those are points that I added to the docs mentioned in the description as this needs to be looked at when finalizing the split (ie removing the dupps from the operator convention, to be done in [OSD-10468](https://issues.redhat.com/browse/OSD-10468) . 
For the very short term, variables are defined the same way in all conventions and we should finish the migration (answering the 'how to') before it becomes troublesome.